lib/adapters/secret_adapter.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'tty-command'
+
+class SecretAdapter
```

Comment:
is the name SecretAdapter specific enough?
K8sSecretAdapter?
then inherit OseSecretAdapter from this class an override where necessary?
I'm suggesting inheriting that way since openshift is based on k8s.
other names if this is not possible: ClusterSecretAdapter?